model/models/threes.py

```diff
@@ -114,16 +123,21 @@ def run_model(
         def_adjustment = opponent_fg_percent_by_cluster / league_fg_percent_by_cluster
 
         #bootstrap resample from FGA data to find normal distribution fo estimated mean FGA per game
-        fga_per_game_est = [np.random.choice(fga_per_game_data, size=len(fga_per_game_data), replace=True).mean() for _ in range(bootstrap_samples)]
-        fga_per_game_est_mean = np.mean(fga_per_game_est)
-        fga_per_game_est_std = np.std(fga_per_game_est)
+        if fga_method == 'simple':
+            fga_per_game_array = fga_per_game_data['SHOT_ATTEMPTED_FLAG'].values
+            fga_per_game_est = [np.random.choice(fga_per_game_array, size=len(fga_per_game_array), replace=True).mean() for _ in range(bootstrap_samples)]
+            fga_per_game_est_mean = np.mean(fga_per_game_est)
+            fga_per_game_est_std = np.std(fga_per_game_est)
+        elif fga_method == 'ewm':
+            fga_per_game_est_mean = fga_per_game_data['SHOT_ATTEMPTED_FLAG'].ewm(span = len(fga_per_game_data)).mean().values[-1]
+            fga_per_game_est_std = fga_per_game_data['SHOT_ATTEMPTED_FLAG'].ewm(span = len(fga_per_game_data)).std().values[-1]
 
         fg3m_s = []
         #simulate n_simulations games
         for _ in trange(n_simulated_games, desc=f'Simulating 3PM outcomes for {player_name} vs {opponent}...'):
 
             #simulate FGA
-            fga_i = np.random.poisson(np.random.normal(fga_per_game_est_mean, fga_per_game_est_std))
+            fga_i = np.random.poisson(max(np.random.normal(fga_per_game_est_mean, fga_per_game_est_std),0))
```

Comment:
So above you're not actually bootstrapping so not fair to draw from a normal distribution to input into the poisson.